dvc/scm/git/__init__.py

```diff
@@ -124,19 +124,33 @@ def ignore(self, path, in_curr_dir=False):
         )
         logger.info(msg)
 
-        self._add_entry_to_gitignore(entry, gitignore, ignore_list)
+        self._add_entry_to_gitignore(entry, gitignore)
 
         self.track_file(os.path.relpath(gitignore))
 
         self.ignored_paths.append(path)
 
     @staticmethod
-    def _add_entry_to_gitignore(entry, gitignore, ignore_list):
-        content = entry
-        if ignore_list:
-            content = "\n" + content
+    def _append_newline_on_not_ending_with_newline(path):
+        if not os.path.exists(path):
+            return
+
+        last = None
+        with open(path, "r", encoding="utf-8") as fobj:
```

Comment:
Don't we check that already in https://github.com/iterative/dvc/pull/2118/files#diff-9fe1cb4148dd9740d77f52dc6d30ba5bL116 ?

Reply:
Yes we do, but if it does not exist we don't create new one.

Reply:
So why are we reading this file again? And then opening it once again to add the newline, and then opening it again to write the rest. Seems suboptimal.

Reply:
True, Ill move the logic

Reply:
We've already read `ignore_list` above and now we read it again to get the last line.